source/iterable-element.d.ts

```diff
@@ -0,0 +1,35 @@
+/**
+Get the element type of an Iterable or AsyncIterable.
+
```

Comment:
> Write (English-prose, not code) about some real-world use-cases where it can be useful. (It can be hard sometimes for users to see where they would use something) - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md

Reply:
elaborated a bit on when this might be handy